File: diploma-server/src/main/kotlin/org/arhor/diploma/util/CustomCollectors.java
package org.arhor.diploma.util;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ArrayNode;

import java.util.stream.Collector;

public final class CustomCollectors {

  private CustomCollectors() {}

  public static Collector<String, ArrayNode, ArrayNode> toArrayNode(final ObjectMapper mapper) {
    return Collector.of(
        mapper::createArrayNode,
        ArrayNode::add,
        ArrayNode::addAll
    );
  }
}

File: diploma-server/src/main/kotlin/org/arhor/diploma/util/CustomCollectors.kt
package org.arhor.diploma.util

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.node.ArrayNode
import java.util.function.BiConsumer
import java.util.function.BinaryOperator
import java.util.function.Supplier
import java.util.stream.Collector

object CustomCollectors {

  @JvmStatic
  fun toArrayNode(mapper: ObjectMapper): Collector<String, ArrayNode, ArrayNode> {
    return Collector.of(
        Supplier {
          mapper.createArrayNode()
        },
        BiConsumer { node: ArrayNode, str: String ->
          node.add(str)
        },
        BinaryOperator { left: ArrayNode, right: ArrayNode ->
          left.addAll(right)
        }
    )
  }
}
